### ISSUE TITLE
[Bug] Siteroot and basePath don't work with routes

### ISSUE THREAD

**Issue by NewBieBR** [react-static/react-static#1141]
### Environment

- System:
    OS: macOS Mojave 10.14.3
    CPU: (8) x64 Intel(R) Core(TM) i7-4870HQ CPU @ 2.50GHz
    Memory: 2.77 GB / 16.00 GB
    Shell: 3.2.57 - /bin/bash
 - Binaries:
    Node: 11.12.0 - /usr/local/bin/node
    Yarn: 1.15.2 - /usr/local/bin/yarn
 - Browsers:
    Chrome: 73.0.3683.103
 - npmPackages:
    react: ^16.8.2 => 16.8.6 
    react-color: ^2.17.0 => 2.17.0 
    react-dom: ^16.8.2 => 16.8.6 
    react-static: ^7.0.0 => 7.0.5 
    react-static-plugin-reach-router: ^7.0.0 => 7.0.0 
    react-static-plugin-sitemap: ^7.0.0 => 7.0.0 
    react-static-plugin-source-filesystem: ^7.0.0 => 7.0.0 
    npmGlobalPackages:
    react-static: 7.0.7

### Steps to Reproduce the problem

- react-static create
> with "basic" template

- In static.config.js
```
import path from "path";

export default {
  siteRoot: "http://localhost:3000",
  basePath: "dist",
  // maxThreads: 1, // Remove this when you start doing any static generation
  plugins: [
    [
      require.resolve("react-static-plugin-source-filesystem"),
      {
        location: path.resolve("src/pages")
      }
    ],
    require.resolve("react-static-plugin-reach-router"),
    require.resolve("react-static-plugin-sitemap")
  ]
};

```
- yarn build && serve . -p 3000

- Go to "http://localhost:3000/dist"

### Expected Behavior

Template example should work normally with its routes

### Problem

It doesn't, console show 404 error for loading ressources, path is incorrect.

**Comment by kevinmanncito** [comment]
Running into the same issue with react-static 7.0.9. These were my steps to reproduce (slightly different than above)

• Created a new react-static basic site with: `react-static create`
• Set `siteRoot: "http://localhost:8000"` and `basePath: "test"` 
• Ran `yarn build`
• Copied the contents of `dist` to a folder named `test`
• Ran a python simple http server from the same level as the test folder `python -m SimpleHTTPServer`

What I want to point out is it looks like the template script tags are the issue:

![Screen Shot 2019-04-23 at 9 01 08 AM](https://user-images.githubusercontent.com/2521298/56591661-5c5bb000-65a6-11e9-8fb5-10d0153113a0.png)

Everything else seems to be using the siteRoot and basePath correctly.

**Comment by Vinnl** [comment]
Same here, can be seen on a live site [here](https://vincenttunru.gitlab.io/blog/).

Console output of failed requests:

![image](https://user-images.githubusercontent.com/4251/56652940-d8c8be80-668c-11e9-89df-a5319860036e.png)

Links [appear to point to the correct page](https://spectrum.chat/react-static/general/why-dont-links-respect-my-basepath-when-clicked~8d09bbcd-4d97-4b03-bf6f-c3fdb6b9de35), but then error out when clicked.

My debugging info:

```
$ npx envinfo --system --npmPackages react* --binaries --npmGlobalPackages react* --browsers
  System:
    OS: Linux 4.15 Ubuntu 18.04.2 LTS (Bionic Beaver)
    CPU: (4) x64 Intel(R) Core(TM) i7-7500U CPU @ 2.70GHz
    Memory: 555.84 MB / 7.52 GB
    Container: Yes
    Shell: 5.4.2 - /bin/zsh
  Binaries:
    Node: 10.15.3 - ~/.nvm/versions/node/v10.15.3/bin/node
    Yarn: 1.15.2 - /usr/bin/yarn
    npm: 6.4.1 - ~/.nvm/versions/node/v10.15.3/bin/npm
  Browsers:
    Chrome: 74.0.3729.108
    Firefox: 66.0.3
  npmPackages:
    react: ^16.8.6 => 16.8.6 
    react-dom: ^16.8.6 => 16.8.6 
    react-ga-donottrack: ^1.0.0 => 1.0.0 
    react-hot-loader: ^4.8.4 => 4.8.4 
    react-markdown: ^4.0.8 => 4.0.8 
    react-static: ^7.0.9 => 7.0.9 
    react-static-plugin-sass: ^7.0.8 => 7.0.8 
    react-static-plugin-typescript: ^7.0.8 => 7.0.8 
    react-syntax-highlighter: ^10.2.1 => 10.2.1 
```

**Comment by jellomoose** [comment]
Having similar issues. basePath is not being applied to the template JS files that are injected into the rendered index.html. I tried digging into the configuration, but was unable to find an appropriate place to add the basePath. Hoping someone can point in the right direction soon...

**Comment by gbrsales** [comment]
I proposed a fix in PR #1196.

**Comment by nddery** [comment]
We're also facing this issue, seems like the templates aren't indeed requested from the `basePath`, but also that the `routeInfo.json` files are requested in "2 `basePath` (note the `/dist/dist` on the `routeInfo.json` path):

![Screen Shot 2019-06-05 at 9 27 49 AM](https://user-images.githubusercontent.com/610207/58960068-5dbaf380-8774-11e9-8e40-94cc72f62cb7.png)

I tried #1196 but it only seemed to change the 404'ing template URL to also have `/dist/dist`:

![Screen Shot 2019-06-05 at 9 32 57 AM](https://user-images.githubusercontent.com/610207/58960338-010c0880-8775-11e9-90a3-bcb033c8e17c.png)

We get this behavior with and without setting `siteRoot` (only setting the `basePath` and/or `stagingBasePath`).
